lisa/tools/kdump.py

```diff
@@ -0,0 +1,155 @@
+        )
+
+    def config_crashkernel_memory(
+        self,
+        crashkernel_memory: str,
+    ) -> None:
+        # For Ubuntu, /etc/default/grub.d/kdump-tools.cfg which is in kdump-tools package,
+        # will override the crashkernel config defined as GRUB_CMDLINE_LINUX_DEFAULT in /etc/default/grub
+        # We can keep the default crashkernel config for Ubuntu distro, no need to set crashkernel
+        posix_os: Posix = cast(Posix, self.node.os)
+        if isinstance(self.node.os, Redhat):
+            posix_os.config_kenel_cmdline(
+                "crashkernel=auto", f"crashkernel={crashkernel_memory}"
+            )
+        elif isinstance(self.node.os, Suse):
+            posix_os.config_kenel_cmdline(
+                '\\"$', f' crashkernel={crashkernel_memory}\\"'
+            )
+
+    def enable_kdump_service(self) -> None:
+        # For Ubuntu, it has kdump-tools.service, when install kdump-tools package, it will enable by default
+        # For Redhat, it has kdump.service which is in kexec-tools package, it will enable by default when install the package?
+        systemctl = self.node.tools[Systemctl]
+        if isinstance(self.node.os, Redhat) or isinstance(self.node.os, Suse):
+            systemctl.enable_service("kdump")
+        elif isinstance(self.node.os, Debian):
+            systemctl.enable_service("kdump-tools")
+
+    def config_nmi_panic(self) -> None:
+        nmi_panic_file = "/proc/sys/kernel/unknown_nmi_panic"
+        if self.node.shell.exists(PurePosixPath(nmi_panic_file)):
+            result = self.node.execute(
+                "sysctl -w kernel.unknown_nmi_panic=1", shell=True, sudo=True
+            )
+            result.assert_exit_code(
+                message="Failed to enable kernel to call panic when it receives a NMI."
+            )
+
+    def check_kdump_loaded(self) -> None:
+        result = self.node.execute(
+            "grep -i crashkernel= /proc/cmdline",
+            shell=True,
+            sudo=True,
+        )
+        result.assert_exit_code(message="crashkernel boot parameter is not present")
+        assert_that(
+            self.node.shell.exists(PurePosixPath(self.kexec_crash))
+        ).described_as(
+            "Kexec crash is not loaded after reboot. "
+            "Please check the configuration settings for kdump and grub."
+        ).is_true()
+
+        if isinstance(self.node.os, Debian):
+            cat = self.node.tools[Cat]
+            result = cat.run(self.kexec_crash)
+            assert_that(result.stdout).described_as(
+                "kexec_crash_loaded is not 1. "
+                "Kexec crash is not loaded after reboot. "
+            ).is_equal_to("1")
+
+    def check_kdump_service(self) -> None:
+        systemctl = self.node.tools[Systemctl]
+        if isinstance(self.node.os, Redhat) or isinstance(self.node.os, Suse):
+            systemctl._check_service_running("kdump")
```

Comment:
The leading `_` means it's a private method. Either make this public, or use another method. Don't call private method out of a class.